vms/rpcchainvm/vm_client.go

```diff
@@ -269,6 +314,17 @@ func (vm *VMClient) BuildBlock() (snowman.Block, error) {
 
 // ParseBlock ...
 func (vm *VMClient) ParseBlock(bytes []byte) (snowman.Block, error) {
+	blkID := hashing.ComputeHash256Array(bytes)
```

Comment:
Is this some known invariant, @aaronbuchwald?

Reply:
I saw you add `vm.ctx.Log.AssertTrue(id != blkID, "Unexpected blockID returned from plugin")` and didn't know if this was a theory you were still confirming.